.github/scripts/verify-code-references.sh

```diff
@@ -0,0 +1,150 @@
+#!/usr/bin/env bash
+# Verify that code references in documentation match actual source code
+# This script extracts CODE_REFERENCE comments from markdown files and validates
+# that the referenced code still exists at the specified line numbers
+
+set -euo pipefail
+
+# Colors for output
+RED='\033[0;31m'
+GREEN='\033[0;32m'
+YELLOW='\033[1;33m'
+NC='\033[0m' # No Color
+
+REPO_ROOT="$(cd "$(dirname "${BASH_SOURCE[0]}")/../.." && pwd)"
+DOCS_DIR="${REPO_ROOT}/website/docs"
+EXIT_CODE=0
+TOTAL_REFS=0
+VALID_REFS=0
+INVALID_REFS=0
+
+echo "Verifying code references in documentation..."
+echo "Repository root: ${REPO_ROOT}"
+echo ""
+
+# Find all markdown files with CODE_REFERENCE comments
+while IFS= read -r doc_file; do
+    # Extract CODE_REFERENCE comments from this file
+    while IFS= read -r line_num; do
+        # Get the actual line content
+        comment_line=$(sed -n "${line_num}p" "$doc_file")
+
+        # Extract file path and line range from comment
+        # Format: <!-- CODE_REFERENCE: path/to/file.rs#L123-L456 -->
+        if [[ $comment_line =~ CODE_REFERENCE:\ *([^#]+)#L([0-9]+)-L([0-9]+) ]]; then
+            file_path="${BASH_REMATCH[1]}"
+            start_line="${BASH_REMATCH[2]}"
+            end_line="${BASH_REMATCH[3]}"
+
+            # Trim whitespace from file_path
+            file_path=$(echo "$file_path" | xargs)
+
+            TOTAL_REFS=$((TOTAL_REFS + 1))
+
+            # Check if the source file exists
+            source_file="${REPO_ROOT}/${file_path}"
+            if [[ ! -f "$source_file" ]]; then
+                echo -e "${RED}✗${NC} Invalid reference in ${doc_file}:${line_num}"
+                echo "  File not found: ${file_path}"
```

Comment:
Does this case cover the file existing but not having permissions?

Reply:
And by the way, this works only for the related repo?  As in, our doc website will. check only references from the rust node repo or can it check any arbitrary repository as long as you configure it?

Reply:
> Does this case cover the file existing but not having permissions?

`-f filename` checks if the file exists. We should not care about the permissions as it would be on code that is supposed to have at least the `read` permissions (code, toml, yaml, etc).